app/experimenter/experiments/tests/test_tasks.py

```diff
@@ -373,3 +374,88 @@ def test_internal_only_does_not_add_comment(self):
         self.mock_bugzilla_requests_post.assert_not_called()
 
         self.assertEqual(Notification.objects.count(), 0)
+
+
+class TestUpdateExperimentStatus(
+    MockRequestMixin, MockNormandyMixin, TestCase
+):
+
+    def setUp(self):
+        super().setUp()
+
+        self.experiment = ExperimentFactory.create_with_status(
+            target_status=Experiment.STATUS_ACCEPTED, normandy_id=1234
+        )
+
+    def test_accepted_experiment_becomes_live_if_normandy_enabled(self):
+        tasks.update_experiment_status()
+        experiment = Experiment.objects.get(normandy_id=1234)
+        self.assertEqual(experiment.status, Experiment.STATUS_LIVE)
+        self.assertTrue(
+            experiment.changes.filter(
+                old_status=Experiment.STATUS_ACCEPTED,
+                new_status=Experiment.STATUS_LIVE,
+            ).exists()
+        )
+
+    def test_accepted_experiment_stays_accepted_if_normandy_disabled(self):
+
+        self.mock_normandy_requests_get.return_value = (
+            self.buildMockSuccessDisabledResponse()
+        )
+        experiment = Experiment.objects.get(normandy_id=1234)
+        tasks.update_experiment_status()
+        self.assertFalse(
+            experiment.changes.filter(
+                old_status=Experiment.STATUS_ACCEPTED,
+                new_status=Experiment.STATUS_LIVE,
+            ).exists()
+        )
+
+    def test_live_experiment_stays_live_if_normandy_enabled(self):
+        experiment = Experiment.objects.get(normandy_id=1234)
+        experiment.status = Experiment.STATUS_LIVE
+        tasks.update_experiment_status()
+        self.assertEqual(experiment.status, Experiment.STATUS_LIVE)
```

Comment:
For this assertion to work you'd have to look up the experiment from the db, because even if the task updated the status of the experiment, the instance you're checking against is the in memory one you got on line 416 and in memory instances don't automatically update to reflect the db.